addons/account_payment/models/payment_provider.py

```diff
@@ -83,8 +83,17 @@ def _setup_payment_method(self, code):
                 'payment_type': 'inbound',
             })
 
+    def _check_existing_payment_method_lines(self, payment_method):
+        existing_payment_method_lines_count =  \
+            self.env['account.payment.method.line'].search_count([('payment_method_id', '=', \
+                payment_method.id)], limit=1)
+        return bool(existing_payment_method_lines_count)
+
     @api.model
     def _remove_provider(self, code):
         """ Override of `payment` to delete the payment method of the provider. """
         super()._remove_provider(code)
+        payment_method = self._get_provider_payment_method(code)
+        if self._check_existing_payment_method_lines(payment_method):
+            raise UserError(_("You cannot uninstall this module as payment method transactions already exist."))
         self._get_provider_payment_method(code).unlink()
```

Comment:
```suggestion
        payment_method = self._get_provider_payment_method(code)
        if self._check_existing_payment_method_lines(payment_method):
            raise UserError(_("You cannot uninstall this module as payment method transactions already exist."))
        super()._remove_provider(code)
        payment_method.unlink()
```

Sorry, I didn't spot that before :sweat_smile: